### PR TITLE
FAPI: Add new function Fapi_GetEsysBlob

### PR DESCRIPTION
**Pull request by JuergenReppSIT**
* The function gets blobs from FAPI objects which can be used to create ESAPI objects. 
* Also an integration test was added.

Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>

### CODE REVIEW

**Comment by AndreasFuchsSIT** [comment]
```
../../src/tss2-fapi/api/Fapi_GetEsysBlob.c:337:44: error: implicit conversion from enumeration type 'enum _FAPI_STATE' to different enumeration type 'enum FAPI_SIGN_STATE' [-Werror,-Wenum-conversion]
        statecase(context->Key_Sign.state, GET_ESYS_BLOB_WAIT_FOR_FLUSH);
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/tss2-fapi/ifapi_macros.h:126:13: note: expanded from macro 'statecase'
        VAR=STATE;
           ~^~~~~
1 error generated.
```

File: src/tss2-fapi/api/Fapi_GetEsysBlob.c [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+#ifndef TEST_FAPI_ASYNC
+    r = Esys_SetTimeout(context->esys, TSS2_TCTI_TIMEOUT_BLOCK);
+    return_if_error_reset_state(r, "Set Timeout to blocking");
+#endif /* TEST_FAPI_ASYNC */
+
+    /* A TPM session will be created to enable object authorization */
+    r = ifapi_session_init(context);
+    return_if_error(r, "Initialize GetEsysBlob");
+
+    r = ifapi_get_sessions_async(context,
```
Comment (AndreasFuchsSIT) [inline src/tss2-fapi/api/Fapi_GetEsysBlob.c:177]: Where is the corresponding `ifapi_get_sessions_finish()` or is the async not even needed ?

File: src/tss2-fapi/fapi_int.h [hunk 2]
```diff
@@ -627,18 +627,33 @@ typedef struct {
  */
 typedef struct {
     bool is_key;                    /**< Entity to be deleted is a key */
-    bool is_persistent_key;         /**< Entity to be deleted is a key */
+    bool is_persistent_key;         /**< Entity to be deleted is a persistent key */
```
Comment (AndreasFuchsSIT) [inline src/tss2-fapi/fapi_int.h:630]: Not for this commit ?

File: src/tss2-fapi/fapi_int.h [hunk 3]
```diff
@@ -627,18 +627,33 @@ typedef struct {
...
     ESYS_TR new_object_handle;
     TPM2_HANDLE permanentHandle;    /**< The TPM permanent handle */
     IFAPI_OBJECT auth_object;       /**< Object used for authentication */
     ESYS_TR auth_index;             /**< The ESAPI handle of the nv authorization object */
-    char *path;                     /**< The name of the file to be deleted */
+    char *path;                     /**< The pathname of the file to be deleted */
```
Comment (AndreasFuchsSIT) [inline src/tss2-fapi/fapi_int.h:635]: same

File: src/tss2-fapi/api/Fapi_GetEsysBlob.c [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+            ifapi_flush_policy_session(context, context->policy.session, r);
+            goto_if_error(r, "Cleanup policy session", error_cleanup);
+
+            /* Flush current object used for blob computation. */
+            r = Esys_FlushContext_Async(context->esys, key_object->handle);
+            goto_if_error(r, "Flush Context", error_cleanup);
+
+            fallthrough;
+
+        statecase(context->Key_Sign.state, GET_ESYS_BLOB_WAIT_FOR_FLUSH);
```
Comment (AndreasFuchsSIT) [inline src/tss2-fapi/api/Fapi_GetEsysBlob.c]: Wrong state

File: test/integration/fapi-get-esys-blobs.int.c [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        LOG_ERROR("Invalid type");
+        goto error;
+    }
+
+    r = Tss2_MU_TPMS_CONTEXT_Unmarshal(data, data_size, &offset, &key_context);
+    goto_if_error(r, "Context unmarshal", error);
+
+    r = Esys_ContextLoad(context->esys, &key_context, &esys_handle);
+    goto_if_error(r, "Context load", error);
+
```
Comment (AndreasFuchsSIT) [inline test/integration/fapi-get-esys-blobs.int.c:135]: Could we add a Esys_Sign or something here so we know that metadata works ?

**Comment by AndreasFuchsSIT** [comment]
One more thing:
```
test/integration/fapi-get-esys-blobs.int.c:104: trailing whitespace.
```

File: src/tss2-fapi/api/Fapi_GetEsysBlob.c (hunk 1 above)
Comment (JuergenReppSIT) [reply src/tss2-fapi/api/Fapi_GetEsysBlob.c:177]: Yes, the async call is not needed. The session is created in ifapi_load_key.

File: src/tss2-fapi/api/Fapi_GetEsysBlob.c [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+    } while ((r & ~TSS2_RC_LAYER_MASK) == TSS2_BASE_RC_TRY_AGAIN);
+
+    return_if_error_reset_state(r, "Entity_GetEsysBlob");
+
+    return TSS2_RC_SUCCESS;
+}
+
+/** Asynchronous function for Fapi_GetEsysBlob
+ *
+ * Deletes a given key, policy or NV index from the system.
```
Comment (AndreasFuchsSIT) [inline src/tss2-fapi/api/Fapi_GetEsysBlob.c:107]: Wrong comments

**Pull request merged**